modules/apps/portal-store/portal-store-s3/src/main/java/com/liferay/portal/store/s3/S3FileCache.java

```diff
@@ -71,26 +71,36 @@ public class S3FileCache {
 	public void cleanUpCacheFiles() {
 		_calledCleanUpCacheFilesCount++;
 
-		if (_calledCleanUpCacheFilesCount <
-				_cacheDirCleanUpFrequency.intValue()) {
+		if ((_calledCleanUpCacheFilesCount <
+				_cacheDirCleanUpFrequency.intValue()) ||
+			_cacheCleanUpRunning) {
 
 			return;
 		}
 
 		synchronized (this) {
-			if (_calledCleanUpCacheFilesCount == 0) {
-				return;
-			}
+			try {
+				_cacheCleanUpRunning = true;
+
+				if (_calledCleanUpCacheFilesCount <
+						_cacheDirCleanUpFrequency.intValue()) {
+
+					return;
+				}
 
-			_calledCleanUpCacheFilesCount = 0;
+				_calledCleanUpCacheFilesCount = 0;
```

Comment:
@mbowerman When using `tryLock`, I think we can remove the if-statement here, and reset the counter after clean-up finishes.

Reply:
Done. Thanks for the quick review!